Tests/PennyTests/GatewayProcessingTests.swift

```diff
@@ -43,11 +43,57 @@ class GatewayProcessingTests: XCTestCase {
                 storage: storage
             )
         }
-        await stateManager._tests_reset()
+        await BotStateManager.shared._tests_reset()
         Task { try await Penny.main() }
-        await manager.waitUntilConnected()
+        await waitForStateManagerShutdownAndDidShutdownSignals()
     }
-    
+
+    func waitForStateManagerShutdownAndDidShutdownSignals() async {
+        /// Wait for the shutdown signal, then send a `didShutdown` signal.
+        /// in practice, the `didShutdown` signal is sent by another Penny that is online.
+        while let possibleSignal = await responseStorage.awaitResponse(
+            at: .createMessage(channelId: Constants.Channels.logs.id)
+        ).value as? Payloads.CreateMessage {
+            if possibleSignal.content?.contains(StateManagerSignal.shutdown.value) == true {
```

Comment:
`== true` is anti-idiomatically redundant; when dealing with the "oops `Bool?` isn't `Bool`" problem in conditionals I generally prefer `(foo.bar?.contains(...) ?? false)` unless doing so is excessively awkward in context. In a case like this, I'd probably go so far as to write `if let content = possibleSignal.content, content.contains(...)`

(Does it _really_ matter, especially in a test? Of course not, but I suspect it's hardly a revelation to anyone by now that I have a very particular idea of "good" Swift style 🙃.)